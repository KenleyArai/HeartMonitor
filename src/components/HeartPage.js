import React, { Component } from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import OutputView from "./OutputView";
import InputView from "./InputView";

export default class HeartPage extends Component<{}> {
  static propTypes = {
    heart_samples: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.number,
        startDate: PropTypes.string
      })
    )
  };

  static defaultProps = {
    heart_samples: [
      {
        value: 80,
        startDate: "2018-03-02T06:12:54.536-0800"
      }
    ]
  };

  render() {
    return (
      <View>
        <OutputView />
        {this.props.heart_samples.map(result => (
          <Text key={result["startDate"]}>
            {"BPM:" + result["value"] + "\t\tDate:" + result["startDate"]}
          </Text>
        ))}
        <InputView />
      </View>
    );
  }
}
