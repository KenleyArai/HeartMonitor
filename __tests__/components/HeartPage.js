import "react-native";
import React from "react";
import HeartPage from "../../src/components/HeartPage";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<HeartPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
