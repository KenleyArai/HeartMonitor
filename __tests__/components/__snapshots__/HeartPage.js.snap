// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Bar
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Graph of BPM
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        VerticalBar
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    BPM:80		Date:2018-03-02T06:12:54.536-0800
  </Text>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Button Skeleton
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        button
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        -1
         
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Button Skeleton
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        button
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        -1
         
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Button Skeleton
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        button
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         
        -1
         
      </Text>
    </View>
  </View>
</View>
`;
